packages/medusa/src/services/pricing.ts

```diff
@@ -115,28 +120,58 @@ class PricingService extends TransactionBaseService<PricingService> {
     }
 
     if (variantPricing.calculated_price !== null) {
-      const taxAmount = Math.round(variantPricing.calculated_price * rate)
+      const taxAmount =
+        this.featureFlagRouter.isFeatureEnabled("tax_inclusive_pricing") &&
```

Comment:
nit: the flag routing can probably be avoided here as `calculated_price_includes_tax` will always be false when the flag is not set.

Reply:
yeah, I considered removing it, but I added it just as an additional guard to be very explicit about the code paths and when they are enabled 😄